utility/src/test/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecoratorTest.kt

```diff
@@ -0,0 +1,38 @@
+package org.oppia.android.util.profile
+
+import android.content.Intent
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.util.profile.CurrentUserProfileIdIntentDecorator.decorateWithUserProfileId
+import org.oppia.android.util.profile.CurrentUserProfileIdIntentDecorator.extractCurrentUserProfileId
+import org.robolectric.annotation.LooperMode
+
+/** Tests for [CurrentUserProfileIdIntentDecorator]. */
+// FunctionName: test names are conventionally named with underscores.
+@Suppress("FunctionName")
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class CurrentUserProfileIdIntentDecoratorTest {
+
+  @Test
+  fun testDecorator_decorateWithProfileId_returnsIntentWithCorrectProfileId() {
```

Comment:
Please update the test names to be more explicit about which API method they're testing. I suggest something like:

``testDecorateWithUserProfileId_emptyProto_addsProtoToIntentAsExtra``

(As an example, not as a replacement for this exact test's name).

That being said, we should be testing both public API methods, and probably the decorate method first (since the extract method depends on it). We should also make sure to test edge cases (e.g. default & missing profile IDs) to make sure the methods do what we expect (and continue staying that way).

Reply:
Thanks Ben have refactored this and also added check for missing profile IDs.